### PR TITLE
[wicketd] switch to a watch channel for installinator reports

### PR DESCRIPTION
**Pull request by sunshowers**
This was originally changed to an unbounded receiver in #3579, but in
the comments to RFD 400 it was pointed out that maybe we shouldn't give
up on backpressure. Switch to a watch channel, since we only care about
the last element anyway.

(This isn't a hugely pressing issue, but I want to include it as a case
study in RFD 400.)

Also treat the receiver being closed as HTTP 410 Gone. We don't do
anything with this at the moment, but could abort installinator on
receiving this signal in the future.

### CODE REVIEW

File: wicketd/src/installinator_progress.rs [hunk 1]
```diff
@@ -195,15 +204,44 @@ impl RunningUpdate {
...
+                // all of its messages.
+                slog::warn!(
+                    log,
+                    "progress receiver is closed -- marking aborted";
+                    "last_seen" => ?report.last_seen,
+                );
+                (
+                    // Don't set the state to Closed here even if this is a
+                    // terminal update, because we want to keep returning
+                    // ReceiverClosed.
```
Comment (jgallagher) [inline wicketd/src/installinator_progress.rs:240]: I don't entirely follow this comment, but I'm having a hard time finding where `Self::Closed` vs `Self::ReportsReceived(_)` is used or reported - can you point me in the right direction?

File: wicketd/src/installinator_progress.rs [hunk 2]
```diff
@@ -72,30 +72,42 @@ impl IprArtifactServer {
...
+                    let (new_state, ret) = RunningUpdate::send_and_next_state(
+                        &self.log, sender, report,
+                    );
+                    *update = new_state;
+                    ret
                 }
                 RunningUpdate::Closed => {
                     // The sender has been closed; ignore the report.
                     *update = RunningUpdate::Closed;
+                    EventReportStatus::Processed
```
Comment (sunshowers) [inline wicketd/src/installinator_progress.rs:105]: This is where the `ReportsReceived` and `Closed` statuses are read.

File: wicketd/src/installinator_progress.rs (hunk 1 above)
Comment (sunshowers) [reply wicketd/src/installinator_progress.rs]: I've expanded the comment -- hopefully this is clearer.

Comment (jgallagher) [reply wicketd/src/installinator_progress.rs]: It is, thank you!

**Review by jgallagher — APPROVED** [review]

**Pull request merged**